Steps, API version and Auto-scrolling

Frank1234 opened the pull request:
Main changes: 
  - added feature: displaying items in steps, going through the steps by clicking.
  - being able to auto-scroll to a certain view when an attached showcase is activated.
  - requirements down to API 15.

Frank1234 committed:
ADD extra card positions

Frank1234 committed:
ADD showing of lists of tips in click-through steps
CHANGE minSdk to 15
CHANGE updated support lib
ADD lambda statements and Java8

Frank1234 committed:
BUGFIX java 8 fix
CHANGE version name change

Frank1234 committed:
CHANGE removed old maven repo

Frank1234 committed:
CHANGE test if submodule updating works

Frank1234 committed:
ADD scroll functionality

Frank1234 committed:
ADD context getter

Frank1234 committed:
ADD dark overlay changes inbetween items

Frank1234 committed:
Some null checks

Frank1234 committed:
ADD examples in example app
ADD examples in readme file
CHANGE some scroll improvements

Frank1234 committed:
ADD builder pattern for ShowCaseStepController

Frank1234 committed:
CHANGE some naming changes

Frank1234 committed:
CHANGE readme

Frank1234 committed:
CHANGE spelling error in ReadMe

dimorinny commented:
Thanks a lot for the contribution.
I need a few days for review this changes.

Frank1234 commented:
Welcome.

Good luck, let me know if you have questions.

You might want to update that Gif btw, it’s flickers a bit after the first item.

> On 21 Aug 2017, at 1:39 PM, Dima Merkurev <notifications@github.com> wrote:
> 
> Thanks a lot for the contribution.
> I need a few days for reviewing this changes.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/dimorinny/show-case-card-view/pull/3#issuecomment-323721859>, or mute the thread <https://github.com/notifications/unsubscribe-auth/ACKloZSwcm_DNsVpEHCJrcnBckzD3Hroks5saWxfgaJpZM4O9O3A>.
>

dimorinny commented on .gitmodules:
```diff
@@ -0,0 +1,3 @@
+[submodule "show-case-card-view"]
```
Why is it necessary?

Frank1234 replied:
can be removed

dimorinny commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -93,5 +93,32 @@
+
+        <View
+            android:layout_width="match_parent"
+            android:layout_height="1000dp"/>
```
Formatting

dimorinny commented on build.gradle:
```diff
@@ -5,7 +5,7 @@ buildscript {
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:2.2.2'
+        classpath 'com.android.tools.build:gradle:3.0.0-beta1'
```
Let's update to beta2?

dimorinny commented on showcasecard/src/main/java/ru/dimorinny/showcasecard/position/BottomCenter.java:
```diff
@@ -0,0 +1,24 @@
+    @Override
+    public PointF getPosition(Activity activity) {
+        switch (ActivityUtils.getOrientation(activity)) {
+            case 2:
```
Use `ORIENTATION_LANDSCAPE` instead hardcode value

Frank1234 replied:
yes

dimorinny commented on showcasecard/src/main/java/ru/dimorinny/showcasecard/step/ShowCaseStep.java:
```diff
@@ -0,0 +1,116 @@
+     * to this item.
+     */
+    @Nullable
+    private View viewToShowCase;
```
I think that you may to use the only position (without this field). Because there is `ViewPosition` class for this case.

Frank1234 replied:
yes, this can be improved. It's been a few weeks, but if I remember correctly, Steps can have two positions: 1. position to scroll to (that centers the view). 2. the position (center of the target view) on the screen to point to after the scroll. ViewPosition only has one coordinate.

Frank1234 replied:
I'll take a look to fix this

dimorinny commented on showcasecard/src/main/java/ru/dimorinny/showcasecard/step/ShowCaseStepDisplayer.java:
```diff
@@ -0,0 +1,241 @@
+     * @param scrollView scrollView to use on all {@link ShowCaseStep}'s that dictate
+     *                   scrolling on activation.
+     */
+    private ShowCaseStepDisplayer(@Nullable Activity activity, @Nullable Fragment fragment, @Nullable ScrollView scrollView) {
```
Maybe let's use different constructors for Activity or Fragment based displayers?
I think it is more explicitly for users.

Frank1234 replied:
yes

Frank1234 replied:
it's private though, the builder has the two constructors, still want it changed?

HeinrichWesson committed:
CHANGE Address a few TODOs

dimorinny commented:
Do you plan to fix this issues?
If not, then I will merge your changes and fix it myself

Frank1234 commented:
Which issue? The Gif? If you could fix it, that would be great.

dimorinny commented:
No, I meant code review comments

Frank1234 commented:
Where can I see the comments? "Reviewers - No reviews" is what I see.

dimorinny commented:
Oh, sorry, I didn't submit my review :)

Frank1234 committed:
CHANGE use of ORIENTATION_LANDSCAPE instead of harcoded position
CHANGE removed git submodule
CHANGE layout style change

Frank1234 committed:
REMOVE removed views from Steps and build a different solution
REMOVE need to provide "scrollTo = true" on steps. Scrolled automatically now when needed.

Frank1234 commented:
I'm done.

dimorinny commented:
It's really great changes, thanks for contributing!
I will publish new version now!

Frank1234 commented:
Welcome, thanks for merging!

What company are you working for?

Regards,
Frank

> On 01 Sep 2017, at 12:00 AM, Dima Merkurev <notifications@github.com> wrote:
> 
> It's really great changes, thanks for contributing!
> I will publish new version now!
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/dimorinny/show-case-card-view/pull/3#issuecomment-326432204>, or mute the thread <https://github.com/notifications/unsubscribe-auth/ACKloSOSGB-jOopbBGpL-WAdiSGPDHVRks5sdy0ZgaJpZM4O9O3A>.
>

dimorinny commented:
I work in Avito company (it is something like eBay but for individuals) as Android / Platform developer (developing some automation tools, CI, etc for our Android developers).

Frank1234 commented:
Hi,

Sounds like an interesting job! Are you enjoying it? I see you have quite a lot of colleagues, wow, 501 - 1k.

I found a bug in the code when running a monkey test on my app. I fixed it and did a pull request.

Sorry about that :)

Regards,
Frank



> On 01 Sep 2017, at 3:15 PM, Dima Merkurev <notifications@github.com> wrote:
> 
> I work in Avito company (it is something like eBay but for individuals) as Android / Platform developer (developing some automation tools and CI for our Android developers).
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/dimorinny/show-case-card-view/pull/3#issuecomment-326577070>, or mute the thread <https://github.com/notifications/unsubscribe-auth/ACKloQuYgAPmxQzQhoAmRruI2I_vH7TSks5seAN7gaJpZM4O9O3A>.
>

dimorinny commented:
Yeah, sometimes it's really interesting :)
Ok, I will look and merge it.
